modules/programs/firefox.nix

```diff
@@ -21,10 +21,42 @@ let
   profilesPath =
     if isDarwin then "${firefoxConfigPath}/Profiles" else firefoxConfigPath;
 
-  nativeMessagingHostsJoined = pkgs.symlinkJoin {
-    name = "home_ff_nmhs";
-    paths = cfg.package.nativeMessagingHosts or cfg.nativeMessagingHosts;
-  };
+  nativeMessagingHosts = let
+    installNativeMessagingPath = "${mozillaConfigPath}/native-messaging-hosts";
+    sourcePackage = pkgs.symlinkJoin {
+      name = "hm_ff_nativeMessagingHosts";
+      paths = cfg.nativeMessagingHosts ++ [
+        (pkgs.runCommand "hm_ff_emptyNativeMessagingHost" { } ''
+          mkdir -p $out/lib/mozilla/native-messaging-hosts
+        '')
+      ] ++ [ cfg.finalPackage ];
+    };
```

Comment:
```suggestion
    sourcePackage = pkgs.symlinkJoin {
      name = "hm_ff_nativeMessagingHosts";
      paths = cfg.nativeMessagingHosts ++ [ cfg.finalPackage ];

      postBuild = ''
        mkdir -p $out/lib/mozilla/native-messaging-hosts
      '';
    };

```

Reply:
Home-manager implements `nix fmt` which has been used. Formatting becomes what it is.

Reply:
This isn't formatting. I am using postBuild of symlinkJoin instead of using `runCommand`

Reply:
This is still relevant